### ISSUE TITLE
[bug]: Escape patterns in regex filtering do not work consistently

### ISSUE THREAD

**Issue by kwiechmann** [gatsbyjs/gatsby#14195]
## Description

```
export const query = graphql`
  query {
    # All of the following regex patterns work when searching for the following URL:
    # http://2.bp.blogspot.com/-Ux3T_02B8JA/UAFVR3RDx_I/AAAAAAAAKzw/4WoEcUecF_Q/s1600/panda+(2).jpg
    #
    # "/^.+\\/panda[+][(]2[)]\\\\.jpg$/"
    # "/^.+\\/panda[+][(]2[)][.]jpg$/"
    remoteCoverImage: file(
      relativePath: {regex: "/^.+\\/panda[+][(]2[)]\\\\.jpg$/"},
    ) {
      ...fluidImage
    }
    
  }
`
```
is unable to find an image. The relativePath regex is deliberately longer so that similar but different URLs: `https://www.foo.com/panda+(2).jpg` (Remote Img) and `panda+(2).jpg` (Local Img) do not unexpectedly match.

### Steps to reproduce

1. Start with a remote Img.
2. Put query into index.js
3. `gatsby develop`
4. Get a NULL error if not found.

### Expected result

Four backslashes should consistently escape characters. The examples all use a combination of double backslashes and square braces. Square braces when the backslashes were not working. In the above code snippet work but are overly confusing. Regex are already complicated enough and should work the way online testers like https://www.regexpal.com do:

Regular Expression: `^.+\/+panda\+\(2\)\.jpg$`
Test String: `http://2.bp.blogspot.com/-Ux3T_02B8JA/UAFVR3RDx_I/AAAAAAAAKzw/4WoEcUecF_Q/s1600/panda+(2).jpg`

Preferred Regular Expression escaped and ready for use within Gatsby: `^.+\\\\/+panda\\\\+\\\\(2\\\\)\\\\.jpg$`

### Actual result

panda+(2).jpg was not found without significant trial and error. Without  @kapral18 's ticket  (https://github.com/gatsbyjs/gatsby/issues/10989) as a clue, might remain unresolved.

### Environment

  System:
    OS: macOS 10.14.4
    CPU: (8) x64 Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz
    Shell: 5.0.2 - /usr/local/bin/bash
  Binaries:
    Node: 10.15.2 - /usr/local/bin/node
    Yarn: 1.15.2 - ~/.yarn/bin/yarn
    npm: 6.9.0 - /usr/local/bin/npm
  Languages:
    Python: 2.7.10 - /usr/bin/python
  Browsers:
    Chrome: 74.0.3729.157
    Safari: 12.1
  npmPackages:
    gatsby: ^2.4.5 => 2.4.5 
    gatsby-image: ^2.0.41 => 2.0.41 
    gatsby-plugin-emotion: ^4.0.6 => 4.0.6 
    gatsby-plugin-offline: ^2.1.0 => 2.1.0 
    gatsby-plugin-remote-images: ^1.0.1 => 1.0.1 
    gatsby-plugin-remove-trailing-slashes: ^2.0.11 => 2.0.11 
    gatsby-plugin-sass: ^2.0.11 => 2.0.11 
    gatsby-plugin-sharp: ^2.0.37 => 2.0.37 
    gatsby-plugin-typography: ^2.2.13 => 2.2.13 
    gatsby-remark-images: ^3.0.11 => 3.0.11 
    gatsby-source-filesystem: ^2.0.34 => 2.0.34 
    gatsby-transformer-json: ^2.1.11 => 2.1.11 
    gatsby-transformer-remark: ^2.3.12 => 2.3.12 
    gatsby-transformer-remote-image: ^2.0.1 => 2.0.1 
    gatsby-transformer-sharp: ^2.1.19 => 2.1.19 
  npmGlobalPackages:
    gatsby-cli: 2.5.15

CC: @prestonso

**Comment by jgosmann** [comment]
I'm encountering the same bug. :/

**Comment by chrish-d** [comment]
Having an issue with this where the `\w` is causing errors in this query;

```
        {
          allStoryblokEntry(filter: {parent_id: {ne: 0}, field_component: {regex: "\b(?:(?!newsIndex|servicesIndex|clientsIndex)\w)+\b"}}) {
            edges {
              node {
                content
                field_component
                lang
                slug
              }
            }
          }
        }
```

> "gatsby-node.js" threw an error while running the createPages lifecycle:
> Syntax Error: Invalid character escape sequence: \w.
>
> GraphQLError: Syntax Error: Invalid character escape sequence: \w.

**Comment by chrish-d** [comment]
Having an issue with this where the `\w` is causing errors in this query;

```
        {
          allStoryblokEntry(filter: {parent_id: {ne: 0}, field_component: {regex: "\b(?:(?!newsIndex|servicesIndex|clientsIndex)\w)+\b"}}) {
            edges {
              node {
                content
                field_component
                lang
                slug
              }
            }
          }
        }
```

> "gatsby-node.js" threw an error while running the createPages lifecycle:
> Syntax Error: Invalid character escape sequence: \w.
>
> GraphQLError: Syntax Error: Invalid character escape sequence: \w.

**Comment by jgosmann** [comment]
Will hopefully fixed by #26592?
